File: Java/Remove-Duplicates-from-Sorted-List.java
/**
 * Definition for ListNode
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    /**
     * @param ListNode head is the head of the linked list
     * @return: ListNode head of linked list
     */
    public static ListNode deleteDuplicates(ListNode head) { 
        // write your code here
        if (head == null || head.next == null) return head;
        ListNode cnt = head;
        ListNode cur = head.next;
        int temp = head.val;
        while (cur != null) {
        	if (cur.val != temp) {
        		temp = cur.val;
        		cnt.next = cur;
        		cnt = cnt.next;
        	}
        	cur = cur.next;
        }
        cnt.next = null;
        return head;
    }
}

File: Python/Remove-Duplicates-from-Sorted-List.py
"""
Definition of ListNode
class ListNode(object):
    def __init__(self, val, next=None):
        self.val = val
        self.next = next
"""
class Solution:
    """
    @param head: A ListNode
    @return: A ListNode
    """
    def deleteDuplicates(self, head):
        # write your code here
        if head is None:
            return head
        p = head
        cur = head.next
        while cur is not None:
            if cur.val != p.val:
                p.next = cur
                p = p.next
            cur = cur.next
        p.next = None
        return head